src/Cli/dotnet/commands/dotnet-restore/RestoreCommandParser.cs

```diff
@@ -147,10 +147,16 @@ private static IEnumerable<CliOption> ImplicitRestoreOptions(bool showHelp, bool
 
             yield return new ForwardedOption<bool>("--no-cache")
             {
-                Description = showHelp ? LocalizableStrings.CmdNoCacheOptionDescription : string.Empty,
-                Hidden = !showHelp
+                Description = string.Empty,
```

Comment:
I think we should still show some description. May be update the string to notify customers that this option is in deprecation path and recommend to use the new option (--no-http-cache) instead.

Reply:
I was thinking we should hide --no-cache from the help output so that no new users use it.  However, we do show --no-http-cache in the help output

Reply:
I agree with the hiding part, but I was wondering how would customers using `--no-cache` currently know about the new option. I see we added an information message to the restore log if customers use the old option https://github.com/NuGet/NuGet.Client/pull/5389/files#diff-b43b40cbcbf3bef2bbf71983856b74834ffa03eb82215efb5a132c42b75f9d18R152. Maybe we should be okay.

On a side note, I think we should also update the docs https://learn.microsoft.com/dotnet/core/tools/dotnet-restore. Please feel free to create an issue in the [NuGet docs repo](https://github.com/NuGet/docs.microsoft.com-nuget/issues). In the past when an option is deprecated for dotnet run there was a blog post explaining the reason for changing the option name https://learn.microsoft.com/dotnet/core/compatibility/sdk/6.0/deprecate-p-option-dotnet-run.

Reply:
Just created a PR for updating the docs : https://github.com/NuGet/docs.microsoft.com-nuget/pull/3163 .